ci.multibranch.Jenkinsfile

```diff
@@ -1,298 +1,519 @@
+        booleanParam(name: 'Clean', defaultValue: false, description: 'clean workspace files'),
         string(name: 'ReleaseName', defaultValue: '', description: ''),
         booleanParam(name: 'Nightly', defaultValue: false, description: 'Push release to nightly'),
-        booleanParam(name: 'PGO', defaultValue: false, description: 'Enable Profile Guided Optimization'),
-        string(name: 'PGOProfiles', defaultValue: 'http://10.180.244.30:8080/ipfs/QmXCSpgk3QMCf229eYpPva22EASeTLAvoWby23LW8MSJ1Z/86.0', description: 'Base URL for fetching PGO Profiles'),
-        booleanParam(name: 'Instrument', defaultValue: false, description: 'Enable an instrumented build for generating profiles for PGO'),
-        string(name: 'Locales', defaultValue: 'de', description: 'Repack for these locales'),
     ]),
 ])
 
-def buildmatrix = [:]
-def postbuildmatrix = [:]
-def signmatrix = [:]
 def shouldRelease = params.ReleaseName?.trim()
-def helpers
-def buildId = new Date().format('yyyyMMddHHmmss')
-def locales = params.Locales ? params.Locales.split(',') : []
 
-node('browser-builder') {
-    checkout scm
+stage('Prepare') {
+    node('browser-builder') {
+        checkout scm
 
-    helpers = load "ci/build-helpers.groovy"
-}
+        triggeringCommitHash = sh(returnStdout: true, script: "git log -n 1 --pretty=format:'%h'").trim()
+
+        if (params.Clean) {
+            sh 'rm -rf mozilla-release/obj*'
+            sh 'rm -rf .cache'
+        }
 
-if (params.Linux64) {
-    def name = 'Linux64'
-    def objDir = 'obj-x86_64-pc-linux-gnu'
-    def artifactGlob = "$objDir/dist/Ghostery-*"
+        download('makecab.exe')
+        download('MacOSX10.12.sdk.tar.bz2')
+        download('MacOSX11.0.sdk.tar.bz2')
 
-    buildmatrix[name] = {
-        node('browser-builder') {
-            helpers.build([
-                name: name,
-                dockerFile: 'Linux.dockerfile',
-                targetPlatform: 'linux',
-                objDir: objDir,
-                artifactGlob: artifactGlob,
-                locales: locales,
-                buildId: buildId,
-                Reset: params.Reset,
-                Clobber: params.Clobber,
-                PGO: params.PGO,
-                Instrument: params.Instrument,
-                PGOProfiles: params.PGOProfiles,
-            ])()
-
-            archiveArtifacts artifacts: "mozilla-release/${objDir}/dist/update/*.mar"
-            archiveArtifacts artifacts: "mozilla-release/${artifactGlob}"
-            archiveArtifacts artifacts: "mozilla-release/browser/config/version*"
-
-            stash name: name, includes: [
-                "mozilla-release/${artifactGlob}",
-            ].join(',')
+        def image = docker.build('ua-build-base', '-f build/Base.dockerfile ./build/ --build-arg user=`whoami` --build-arg UID=`id -u` --build-arg GID=`id -g`')
+
+        image.inside() {
+
+            sh 'npm ci'
+
+            sh 'rm -rf mozilla-release'
```

Comment:
The directory is always removed (not like outside whether there is the "params.Clean" flag). I assume, it is intended, especially if the operations are fast. But I don't understand what goes on in the "image.inside" block:

It seems to check out a clean version of the code with all patches applied and API keys for Google. And in the end, it stashes something that is relevant for MacOS (it it the installer?).

Reply:
this is a symlink, does not delete anything